portfolio/src/main/java/com/google/sps/servlets/CommentServlet.java

```diff
@@ -1,91 +1,18 @@
-import com.google.appengine.api.datastore.DatastoreServiceFactory;
-import com.google.appengine.api.datastore.Entity;
-import com.google.gson.Gson;
-import com.google.sps.data.Comment;
-
-import java.io.IOException;
-import javax.servlet.annotation.WebServlet;
 import javax.servlet.http.HttpServlet;
 import javax.servlet.http.HttpServletRequest;
-import javax.servlet.http.HttpServletResponse;
-
-import java.time.LocalDateTime;
-import java.util.ArrayList;
-import java.util.List;
-
-/** Servlet that handles getting and posting comment content. */
-@WebServlet("/comment")
-public final class CommentServlet extends HttpServlet {
-  private final List<Comment> comments = new ArrayList<>();
-  
-  @Override
-  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    String json = convertToJsonUsingGson(this.comments);
-
-    // Send the JSON as the response
-    response.setContentType("application/json;");
-    response.getWriter().println(json);
-  }
-
-  private String convertToJsonUsingGson(List<Comment> comments) {
-    Gson gson = new Gson();
-    String json = gson.toJson(comments);
-    return json;
-  }
-
-  @Override
-  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    // Get comments from the form
-    String commenter = getParameter(request, /*name=*/"commenter", /*defaultValue=*/"");
-    String content = getParameter(request, /*name=*/"content", /*defaultValue=*/"No comments");
-    long timestamp = System.currentTimeMillis();
-    // TODO: validate request parameters
-
-    // Modify the state of server.
-    addComment(commenter, content);
-
-    Entity commentEntity = new Entity("Comment");
-    commentEntity.setProperty("commenter", commenter);
-    commentEntity.setProperty("content", content);
-    commentEntity.setProperty("timestamp", timestamp);
-
-    DatastoreService datastoreService = DatastoreServiceFactory.getDatastoreService();
-    datastoreService.put(commentEntity);
-
-    // Redirect back to the HTML page.
-    response.sendRedirect("/index.html");
-  }
-
-  private void addComment(String name, String comment) {
-    comments.add(new Comment(name, comment, LocalDateTime.now()));
-  }
 
-  /**
-   * @return the value of parameter with the {@code name} in the {@code request}
-   *         or returns {@code defaultValue} if that parameter does not exist.
-   */
-  private String getParameter(HttpServletRequest request, String name, String defaultValue) {
-    String value = request.getParameter(name);
-    if (value == null) {
-      return defaultValue;
+public class CommentServlet extends HttpServlet {
+    /**
+     * @return the value of parameter with the {@code name} in the {@code request}
+     *         or returns {@code defaultValue} if that parameter does not exist.
+     */
+    protected String getParameter(HttpServletRequest request, String name, String defaultValue) {
```

Comment:
I don't think this util method is worth one extra layer of inheritance. This method could be static any way. So maybe a static util method instead of a base class impl.

Reply:
Fixed